game-core/src/main/java/games/strategy/triplea/delegate/battle/grouptarget/Aa.java

```diff
@@ -0,0 +1,70 @@
+package games.strategy.triplea.delegate.battle.grouptarget;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.attachments.TechAbilityAttachment;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.Matches;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lombok.Builder;
+import lombok.NonNull;
+import org.triplea.java.collections.CollectionUtils;
+
+@Builder
+public class Aa {
+
+  private @NonNull final Collection<Unit> aaUnits;
+  private @NonNull final List<String> aaTypes;
+  private @NonNull final GamePlayer hitPlayer;
+  private @NonNull final Collection<Unit> attackableUnits;
+  private final boolean defending;
+  private @NonNull final GameData gameData;
+
+  public List<FiringGroup> getFiringGroups() {
+    final List<FiringGroup> aaGroupsAndTargets = new ArrayList<>();
+
+    for (final String aaType : aaTypes) {
+      final Collection<Unit> aaTypeUnits =
+          CollectionUtils.getMatches(aaUnits, Matches.unitIsAaOfTypeAa(aaType));
+      getGroupsAndTargets(aaGroupsAndTargets, aaType, aaTypeUnits);
```

Comment:
This function should have a different name since it has a set effect and not just a getter.

Maybe updateGroupsAndTargets()?